python/mpi_cmake_modules/utils.py

```diff
@@ -0,0 +1,115 @@
+"""utils
+Utility functions for creating formatting scripts.
+
+License BSD-3-Clause
+Copyright (c) 2019, New York University and Max Planck Gesellschaft.
+"""
+
+import argparse
+import sys
+from os import path, walk
+
+try:
+    from shutil import which
+except:
+    import distutils.spawn
+
+    ## This ensure the compatibility Python2 vs Python3
+    which = distutils.spawn.find_executable
+
+
+def test_valid_file(filename, extensions):
+    """Test if the input file exists and is of one of the provided extension.
+
+    Args:
+        filename (str): Path to the file to test.
+
+        extensions (str): iterable of accepted extensions.
+
+    Returns:
+        True if the file exits and ends with one of the extension.
+
+        False otherwise.
+    """
+    if path.isfile(filename) and any(
+        [filename.endswith(suffix) for suffix in extensions]
+    ):
+        return True
+    else:
+        return False
+
+
+def get_absolute_path(file_or_directory):
+    """Get the absolute path of a given path and check its existence
+
+    Check if file_or_directory is an existing file or directory,
+    if so, returns it. Otherwise, assumes it is a relative path,
+    which it upgrades to an absolute path, this it returns. If the
+    upgraded path does not correspond to an existing file or directory,
+    returns None
+
+    Args:
+          file_or_directory: str `--` a relative or an absolute path
+
+    Returns:
+          an absolute existing path (str) or None
+    """
+    if path.isfile(file_or_directory) or path.isdir(file_or_directory):
+        return file_or_directory
```

Comment:
I think this could be removed.  If the path is already absolute, calling `abspath` on it should simple leave it unchanged.